utils/polus-bfio-util/bfio/bfio.py

```diff
@@ -55,194 +82,450 @@ class BioReader():
+                for i in range(ifdsize):
+                    tag, datatype, count, data = struct.unpack(">HHLL", fhandle.read(12))
+                    if tag == 256:
+                        if datatype == 3:
+                            width = int(data / 65536)
+                        elif datatype == 4:
+                            width = data
+                        else:
+                            raise ValueError("Invalid TIFF file: width column data type should be SHORT/LONG.")
+                    elif tag == 257:
+                        if datatype == 3:
+                            height = int(data / 65536)
+                        elif datatype == 4:
+                            height = data
+                        else:
+                            raise ValueError("Invalid TIFF file: height column data type should be SHORT/LONG.")
+                    if width != -1 and height != -1:
+                        break
+                if width == -1 or height == -1:
+                    raise ValueError("Invalid TIFF file: width and/or height IDS entries are missing.")
+            # handle little endian Tiff
+            elif size >= 8 and head.startswith(b"\x49\x49\x2a\x00"):
+                offset = struct.unpack('<L', head[4:8])[0]
+                fhandle.seek(offset)
+                ifdsize = struct.unpack("<H", fhandle.read(2))[0]
+                for i in range(ifdsize):
+                    tag, datatype, count, data = struct.unpack("<HHLL", fhandle.read(12))
+                    if tag == 256:
+                        width = data
+                    elif tag == 257:
+                        height = data
+                    if width != -1 and height != -1:
+                        break
+                if width == -1 or height == -1:
+                    raise ValueError("Invalid TIFF file: width and/or height IDS entries are missing.")
+            # handle little endian BigTiff
+            elif size >= 8 and head.startswith(b"\x49\x49\x2b\x00"):
+                bytesize_offset = struct.unpack('<L', head[4:8])[0]
+                if bytesize_offset != 8:
+                    raise ValueError('Invalid BigTIFF file: Expected offset to be 8, found {} instead.'.format(offset))
+                offset = struct.unpack('<Q', head[8:16])[0]
+                fhandle.seek(offset)
+                ifdsize = struct.unpack("<Q", fhandle.read(8))[0]
+                for i in range(ifdsize):
+                    tag, datatype, count, data = struct.unpack("<HHQQ", fhandle.read(20))
+                    if tag == 256:
+                        width = data
+                    elif tag == 257:
+                        height = data
+                    if width != -1 and height != -1:
+                        break
+                if width == -1 or height == -1:
+                    raise ValueError("Invalid BigTIFF file: width and/or height IDS entries are missing.")
+
+        return width, height
```

Comment:
What about big endian BigTiff?

Reply:
I think because I reviewed your PRs out of order  have caused a merge conflict. Sorry. Should be simple to update. Let me know when done and then I'll merge.